Client/Frontend/Browser/DefaultBrowserIntroManager.swift

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import BraveShared
+
+private typealias IntroPrefs = Preferences.DefaultBrowserIntro
+
+struct DefaultBrowserIntroManager {
+    /// This function should be called when the app is initialized.
+    /// It determines whether we should show default browser intro popup or not and sets corresponding preferences accordingly.
+    /// Returns true if the popup should be shown.
+    @discardableResult
+    static func prepareAndShowIfNeeded(isNewUser: Bool, launchDate: Date = Date()) -> Bool {
+        // Empty new user value means it is the first time this code is called after
+        // the default intro callout was introduced. Afterwards it should be never nil.
+        if IntroPrefs.isNewUser.value == nil {
```

Comment:
This logic is confusing because it almost seems like `IntroPrefs.isNewUser` is not used really, when in fact its used for subsequent launches based on whether or not the first launch with this new version is a new user vs an existing user. 

I'd recommend either some name changes around arguments/pref names and/or some additional comments around the logic/flow

Reply:
had similar concerns when i started writing tests for it, will try to improve it